lib/Connection.php

```diff
@@ -39,22 +40,23 @@ abstract class Connection extends Component {
     /** @var string $name -- database name (redundant, but mandatory) */
     public $name;
     /** @var string|null $migrationPath -- directory of migration files if not the default (`/migrations`)*/
-    public $migrationPath;
+    public $migrationPath = null;
     /** @var string|null $migrationTable -- name of the migration table if not the default (`migration`) */
     public $migrationTable;
 
-    /** @var PDO $pdo -- the original PDO connection */
-    public $pdo;
+    /** @var PDO|null $_pdo -- the original PDO connection */
+    public $_pdo = null;
```

Comment:
Private variable? If not, then there is a semantic conflict with `getPdo()`...

Reply:
My mistake: I wrote _private_ but what I meant was _not public_ (which includes _protected_ visibility). As for the said semantic conflict, it comes from the fact that `$_pdo` could be accessed from the outside of the class, thus competing with `getPdo()` which is supposed to set it and should be the only publicly visible of the two.